operator/operator.py

```diff
@@ -242,21 +302,37 @@ def check_delete(self):
                 raise
 
     def check_update_namespace(self, namespace):
+        config = UserNamespaceConfig.get(self.config_name) if self.config_name else None
         namespace_name = namespace.metadata.name
         updated = False
 
         if not namespace.metadata.annotations:
-            self.logger.info('setting namespace requester annotation')
-            namespace.metadata.annotations = {'openshift.io/requester', self.user_name}
-            updated = True
-        elif self.user_name != namespace.metadata.annotations.get('openshift.io/requester', ''):
-            self.logger.info('setting namespace requester annotation')
-            namespace.metadata.annotations['openshift.io/requester'] = self.user_name
+            self.logger.info('setting namespace annotations')
+            namespace.metadata.annotations = {
+                'openshift.io/description': self.description,
+                'openshift.io/displayName': self.display_name,
+                'openshift.io/requester': self.user_name,
+            }
             updated = True
+        else:
+            if self.description != namespace.metadata.annotations.get('openshift.io/description', ''):
+                self.logger.info('setting namespace description annotation')
+                namespace.metadata.annotations['openshift.io/description'] = self.description
+                updated = True
+            if self.display_name != namespace.metadata.annotations.get('openshift.io/display-name', ''):
+                self.logger.info('setting namespace display-name annotation')
+                namespace.metadata.annotations['openshift.io/display-name'] = self.display_name
+                updated = True
+            if self.user_name != namespace.metadata.annotations.get('openshift.io/requester', ''):
+                self.logger.info('setting namespace requester annotation')
+                namespace.metadata.annotations['openshift.io/requester'] = self.user_name
+                updated = True
```

Comment:
Not sure if it's worth it but you could DRY here:

```python
for k,v in {"description": "description", "display_name": "display-name", "user_name": "requester"}:
  annotation = 'openshift.io/%s' %(v)
  if getattr(self, k) != namespace.metadata.annotations.get(annotation, ''):
    self.logger.info('setting namespace %s annotation' %(k))
    namespace.metadata.annotations[annotation] = getattr(self, k)
    updated = True
```

Makes me think we should maybe use the same name to avoid maintaining a map for translating the names. Less of a mental load too.
Which would become

```python
from slugify import slugify
for i in ["description", "display_name", "requester"]:
  annotation = 'openshift.io/%s' %(slugify(i))
  if getattr(self, i) != namespace.metadata.annotations.get(annotation, ''):
    self.logger.info('setting namespace %s annotation' %(i))
    namespace.metadata.annotations[annotation] = getattr(self, i)
    updated = True
```

Reply:
Hrm... the `user_name` to `requester` seems not quite right. When a user namespace is created for a user the user didn't "request" it. We use the `openshift.io/requester` annotation, because that is how OpenShift tracks ownership of namespaces, but the semantics aren't quite right. A UserNamespace has a User and that User has a name, so `user_name`.

I think it is right on the threshold of needing the sort of DRY refactor you suggest... I think maybe instead going with a utility function for setting annotations, maybe something like?

```
  if set_annotations(
    namespace,
    {
      'openshift.io/requester': self.user_name,
      'openshift.io/description': self.description,
      'openshift.io/display-name': self.display_name,
    }
  ):
    updated = True
```

Reply:
sounds good